name: CI of releases

on:
  push:
    branches: [ 'release/**' ]

jobs:


  darker:
    uses: ./.github/workflows/darker.yml


  pytest:
    uses: ./.github/workflows/pytest.yml


  npm-test:
    uses: ./.github/workflows/npm-test.yml


  deploy-to-beta:
    runs-on: ubuntu-22.04
    needs:
      - darker
      - pytest
      - npm-test
    if: |
      always() &&
      (needs.darker.result == 'skipped' || needs.darker.result == 'success') &&
      (needs.pytest.result == 'skipped' || needs.pytest.result == 'success') &&
      (needs.npm-test.result == 'skipped' || needs.npm-test.result == 'success')
    steps:
    - uses: actions/checkout@v4
    - name: deploy to beta
      run: |
        echo 'TODO, for now deploy manually on kf.beta.kbtdev'


  on-failure:
    if: |
      always() &&
      (needs.darker.result == 'failure' || needs.darker.result == 'timed_out' ||
      needs.pytest.result == 'failure' || needs.pytest.result == 'timed_out' ||
      needs.npm-test.result == 'failure' || needs.npm-test.result == 'timed_out' ||
      needs.deploy-to-beta.result == 'failure' || needs.deploy-to-beta.result == 'timed_out')
    needs:
      - darker
      - pytest
      - npm-test
      - deploy-to-beta
    uses: './.github/workflows/zulip.yml'
    with:
      topic: "Github Actions (releases)"
      secrets: inherit
      content: ":warning: @*devs*, [run #${{github.run_number}}](${{github.server_url}}/${{github.repository}}/actions/runs/${{github.run_id}}) failed on `${{ github.head_ref || github.ref_name }}` at [\"$COMMIT_TITLE\"](${{ github.event.compare }})"
